### PR TITLE
Test events when POST is redirected

### PR DESCRIPTION
**Pull request by hallvors**
We lacked a test for POSTing content with XHR to a URL that redirects, here it is

### CODE REVIEW

**Comment by hallvors** [comment]
@cdumez - @annevk told me to ping you and let you know I'm checking in a XHR test that fails in Safari. It seems to not pass POST data through a 307 redirect. Just FYI ;)

**Comment by hallvors** [comment]
Travis failure is related to code I haven't changed in this PR so I hope it can be merged anyway..?

File: XMLHttpRequest/send-redirect-post-upload.htm [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                "xhr onreadystatechange 2",
+                /* we expect no onreadystatechange readyState=3 event because there is no loading content */
+                "xhr onreadystatechange 4",
+                "xhr load 4",
+                "xhr loadend 4"
+            ];
+        }
+
+        test.step(function()
+        {
```
Comment (jdm) [inline XMLHttpRequest/send-redirect-post-upload.htm]: nit: move this to the previous line.

File: XMLHttpRequest/send-redirect-post-upload.htm [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            
+            xhr.onreadystatechange = test.step_func(function() {
+                if(xhr.readyState >= xhr.HEADERS_RECEIVED) {
+                    assert_equals(xhr.status, 200, "JS never gets to see the 30x status code");
+                }
+                actual.push("xhr onreadystatechange " + xhr.readyState);
+            });
+            
+            xhr.onload = test.step_func(function(e)
+            {
```
Comment (jdm) [inline XMLHttpRequest/send-redirect-post-upload.htm]: nit: move this to the previous line

File: XMLHttpRequest/send-redirect-post-upload.htm [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                actual.push("xhr load " + xhr.readyState);
+            });
+
+            xhr.onloadend = function(e)
+            {
```
Comment (jdm) [inline XMLHttpRequest/send-redirect-post-upload.htm]: nit: move this to the previous line

File: XMLHttpRequest/send-redirect-post-upload.htm [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                }
+                actual.push("xhr onreadystatechange " + xhr.readyState);
+            });
+            
+            xhr.onload = test.step_func(function(e)
+            {
+                actual.push("xhr load " + xhr.readyState);
+            });
+
+            xhr.onloadend = function(e)
```
Comment (jdm) [inline XMLHttpRequest/send-redirect-post-upload.htm]: Let's make this test.step_func instead, and remove the step call inside of it.

**Comment by jdm** [comment]
This looks fine apart from the small nits.

File: XMLHttpRequest/send-redirect-post-upload.htm [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                "upload loadend 1",
+                "xhr onreadystatechange 2",
+                /* we expect no onreadystatechange readyState=3 event because there is no loading content */
+                "xhr onreadystatechange 4",
+                "xhr load 4",
+                "xhr loadend 4"
+            ];
+        }
+        test.step(function()
+        {
```
Comment (jdm) [inline XMLHttpRequest/send-redirect-post-upload.htm]: nit: move this to the previous line.

File: XMLHttpRequest/send-redirect-post-upload.htm [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+                    assert_equals(xhr.status, 200, "JS never gets to see the 30x status code");
+                }
+                actual.push("xhr onreadystatechange " + xhr.readyState);
+            });
+            xhr.onload = test.step_func(function(e)
+            {
+                actual.push("xhr load " + xhr.readyState);
+            });
+            xhr.onloadend = test.step_func(function(e)
+            {
```
Comment (jdm) [inline XMLHttpRequest/send-redirect-post-upload.htm]: nit: move this to the previous line.

File: XMLHttpRequest/send-redirect-post-upload.htm [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+                actual.push("xhr loadstart " + xhr.readyState);
+            });
+            xhr.onreadystatechange = test.step_func(function() {
+                if(xhr.readyState >= xhr.HEADERS_RECEIVED) {
+                    assert_equals(xhr.status, 200, "JS never gets to see the 30x status code");
+                }
+                actual.push("xhr onreadystatechange " + xhr.readyState);
+            });
+            xhr.onload = test.step_func(function(e)
+            {
```
Comment (jdm) [inline XMLHttpRequest/send-redirect-post-upload.htm]: nit: move this to the previous line.

**Comment by hallvors** [comment]
Ops, sorry. Didn't see those last comments.

**Pull request merged**